pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,347 @@
+"""
+
+from __future__ import absolute_import, print_function
+
+import click
+
+from pywbem import Error
+
+from .pywbemcli import cli
+from ._common import warning_msg, CMD_OPTS_TXT, GENERAL_OPTS_TXT, \
+    SUBCMD_HELP_TXT
+from ._common_options import add_options, help_option
+from ._click_extensions import PywbemcliGroup, PywbemcliCommand
+
+# NOTE: A number of the options use double-dash as the short form.  In those
+# cases, a third definition of the options without the double-dash defines
+# the corresponding option name, ex. 'include_qualifiers'. It should be
+# defined with underscore and not dash
+
+# Issue 224 - Exception in prompt-toolkit with python 2.7. Caused because
+# with prompt-toolkit 2 + the completer requires unicode and click_repl not
+# passing help as unicode in options as unicode
+# NOTE: Insure that all option help attributes are unicode to get around this
+#       issue
+
+
+@cli.group('statistics', cls=PywbemcliGroup, options_metavar=GENERAL_OPTS_TXT,
+           subcommand_metavar=SUBCMD_HELP_TXT)
+@add_options(help_option)
+def statistics_group():
+    """
+    Command group for WBEM server statistics.
+
+    This command group defines commands to control the display of statistical
+    data by pywbem in conjunction with the ``--timestats`` general option which
+    enables the gathering and display of statistics.
+    Since statistical information is gathered both in the pywbem client and
+    the server, there are controls for both elements.
+
+    This command group allows enabling the gathering of statistics in a
+    WBEM server by setting the CIM_ObjectManager "GatherStasticalData" property.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'pywbemcli --help') can also be specified before the
+    'statistics' keyword.
+    """
+    pass  # pylint: disable=unnecessary-pass
+
+
+@statistics_group.command('reset', cls=PywbemcliCommand,
+                          options_metavar=CMD_OPTS_TXT)
+@add_options(help_option)
+@click.pass_obj
+def statistics_reset(context):
+    """
+    Reset the values in the client local statistics.
+
+    This command does not reset statistics in the WBEM server but
+    it does reset the statistics kept in pywbemcli on the
+    ``WBEMServerResponseTime``.
+    """
+    # pylint: disable=too-many-function-args
+    context.execute_cmd(lambda: cmd_statistics_reset(context))
+
+
+@statistics_group.command('server-on', cls=PywbemcliCommand,
+                          options_metavar=CMD_OPTS_TXT)
+@add_options(help_option)
+@click.pass_obj
+def statistics_server_on(context):
+    """
+    Enable statistics on current server.
+
+    This may fail if the server does not manage statistics or does not allow
+    a client to modify the state of statistics gathering.
+
+    This does not impact pywbem local client statistics gathering. See the
+    --timestats general option.
```

Comment:
Let's just use "client statistics" instead of "pywbem local client statistics". Also, it is not clear whether "does not impact" refers to the failure described in the previous sentence, or to the normal successful operation. How about:

"This command does not control client statistics gathering; for that, see the --timestats general option."

Reply:
DONE